### ISSUE TITLE
Groups - reassign owner does not find new friends unless you clear the cache in admincp.

### ISSUE THREAD

**Issue by PhpFoxJohnJr** [PHPfox-Official/phpfox-v4-issues#2412]
## What happened?
I create a new user through an invite.  Signed up the user and the user becomes friends with the sender of the invite...check it out and it works.  I went to the profile page of both and they show each other as friends.  Then I went to a group page as the owner and edit the group page under the manage menu.  Went to the invite tab and it shows the new user I just made as a friend to invite.  Got out of edit mode and went back to the front page of this specific group page and went the reassign owner link.  I begin to type in the name of the new user and it does not show up as a friend to reassign this.  I did this like 10 different users and although they show up in other places...the people do not show up when the code uses module name='friend.search-small' so I then changed the code to use module name='friend.search' which is the code that is used in the tab invite in managed groups.  When I did that...the new user/friend comes up immediately.  Of course, you can't use that code since the variable are set different but it shows up without refreshing the cache in the admincp.  Which is the only way I could get the name to come up before and sometimes the shortcut cache didn't make it work as I had to use the original cache button to get instant results.  So it comes down to the code for friend.search-small is not clearing the cache before it checks and displays your friends.

So I spent the day setting up Redis...thinking I could just get rid of the cache issue.  Well, I think I set up Redis correctly as my memory usage went up from 12 gigs to 43gig and there are only two folders now in the PF.Base/file/cache called template and twig which use to be at least 20 folders and some files.

Anyways, besides not seeing any improvement according to gtmetrix and pingdom I still have the issue where I have to clear the cache button to get the person to show up in the friend box when you reassign a group page owner.

Phpfox new version 4.6.6
centos 7 with plesk.

**Comment by PhpFoxJohnJr** [comment]
I need a quick fix for this as I about to launch this site and I found this as I was doing the help files.  Just tell me what code I need to slap in as this feature in the most important for this site and I cannot move forward without it.  I tried clear_cache code as well some pieces of code I found in other areas but no luck.

**Comment by harrison05** [comment]
We saw this issue, it seems occurs when:
- An invited friend is signed up and become friend with invitation's owner 
Or 
- Setting "On Signup New Friend" is enabled. 

The root cause is system doesn't clear cache after auto adding a new friend for the new user.
We will check and fix in Core.

BTW, this is a quick solution for you:  add below codes

```php
//remove friends list cache
$this->cache()->remove('friend_build_cache_data_' . $iId);
$this->cache()->remove('friend_build_cache_data_' . $iFriendId);
``` 
after wherever system call this function: 
**_Phpfox::getService('friend.process')->updateFriendCount(...)_**

_Note: variables **$iId** **$iFriendId** can be changed to others based on codes in each other file._

Regards.

**Comment by PhpFoxJohnJr** [comment]
Darn...I even tried $this->cache()->remove but I didn't have friend_build_cache_data_ .....I would have never figured that part out in a million years...you are a genius!  Thanks it worked perfectly!

**Comment by PhpFoxJohnJr** [comment]
So you guys don't mark what version this was fixed on anymore as I know you use to?  Here is the list for 4.7
https://docs.phpfox.com/display/FOX4MAN/phpFox+4.7.0
Was it fixed in 4.7?

**Comment by harrison05** [comment]
Yes, it was fixed in v4.7.0.
